clang/lib/FrontendTool/ExecuteCompilerInvocation.cpp

```diff
@@ -42,6 +47,14 @@ CreateFrontendBaseAction(CompilerInstance &CI) {
   StringRef Action("unknown");
   (void)Action;
 
+  auto UseCIR = CI.getFrontendOpts().UseClangIRPipeline;
```

Comment:
Coding standard doesn't allow use of 'auto' here, only when the actual type of the variable is spelled on the RHS side (see next 2 lines too).

Reply:
Should this be the case for actual lib/CIR/CodeGen too? MLIR uses `auto` extensively (perhaps because value semantics is common land there) and we have been doing that as well for CIRGen and passes - it's going to be a massive change for us, so we it'd be good to know if that should apply everywhere or only into the areas of Clang isolated from MLIR.

Reply:
Thats a question I think for @AaronBallman .  This DOES live in the 'clang' directory, so I would think this should all follow the Clang coding standards, but I can see the justification for using the MLIR standard, same as the capitalization (@lanza ).  

There WAS some discussion elsewhere, but I cant find the thread, so I dont know what was agreed upon.

Reply:
Right, it's a different discussion than naming. @joker-eph: what's the `auto` policy for MLIR, is it different from LLVM/Clang? I'm also having trouble to find the previous discussions, let me search a bit more.

Reply:
MLIR isn't meant to differ from LLVM/Clang.

> Coding standard doesn't allow use of 'auto' here, only when the actual type of the variable is spelled on the RHS side (see next 2 lines too).

Nit: technically the coding standard does not say that, I believe you're mentioning a sufficient condition, not a necessary one, see https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable

> Use auto if and only if it makes the code more readable or easier to maintain.

Reply:
> Nit: technically the coding standard does not say that, I believe you're mentioning a sufficient condition, not a necessary one, see https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable
> 
> > Use auto if and only if it makes the code more readable or easier to maintain.

Yup. FWIW, the rule of thumb we use in Clang is that "readable" means "type is spelled out in the initialization somewhere or is otherwise painful to spell but is contextually quite obvious (e.g., use of iterators)" + "any time the code reviewer asks to switch away from `auto`"